### DISCUSSION TITLE
can't evaluate field IsMultilingual in type interface {}  error after update doks-core  from 1.7.0  to  1.8.0

### DISCUSSION THREAD

**Discussion by cnichte** [thuliteio/doks#1307] [Q&A]
Hi, I updated  a clean install of thulite  from `@thulite/doks-core  ^1.7.0  →   ^1.8.0`.

> Ps. This error occurs also on a complete new install with <https://docs.thulite.io/getting-started/>
> I use Hugo v0.121.2

```bash
npm run dev
```
I get an error:

* ERROR render of "page" failed: `node_modules/@thulite/core/layouts/_default/baseof.html:17:7`:
* execute of template failed: template: `blog/single.html:17:7:` executing `blog/single.html`
* at `<partial "footer/script-footer" .>`:
* error calling partial: `node_modules/@thulite/doks-core/layouts/partials/footer/script-footer.html:23:127`: * execute of template failed: template: `partials/footer/script-footer.html:23:127`:
* executing `partials/footer/script-footer.html`
* at `<hugo>: can't evaluate field IsMultilingual in type interface {}`

in `node_modules/@thulite/doks-core/layouts/partials/footer/script-footer.html` is this line:

```bash
  {{ partial "footer/esbuild-template" (dict "src" $flexsearchLangConfigPath "load" "async" "transpile" false "isMultilingual" hugo.IsMultilingual "searchLimit" $searchLimit) -}}
```

I compared my `layouts/blog/single.html` with the `/node_modules/@thulite/doks-core/layouts/blog/single.html`. But there doesn't seem to be any relevant difference.

What could cause this?
Seems there is a field can't evaluate field `IsMultilingual` missing?

Any help is welcome :-) Thank you!

**Comment by h-enk** [comment]
I guess this is no longer relevant — https://github.com/orgs/thuliteio/discussions/493#discussioncomment-11211731

**Reply by cnichte** [reply]
Unfortunately this is still relevant, that was after the update, when I updated again with the latest updates - or so :-)

**Reply by cnichte** [reply]
I made the transition to thulite with the versions on the left  `docs-core 1.7.0`. The error occurs after upgrade like so:

```bash
Patch   Backwards-compatible bug fixes
❯ ◉ @thulite/seo                ^2.4.0  →   ^2.4.1

Minor   Backwards-compatible features
  ◉ @thulite/doks-core          ^1.7.0  →   ^1.8.0
  ◉ @thulite/inline-svg         ^1.1.0  →   ^1.2.0
```

But the error also occurs with a complete fresh and clean default install of thulite from latest version `docs-core 1.8.0`.

**Reply by h-enk** [reply]
Hmm, I cannot reproduce your issue using `npm create thulite@latest` (it works OK for me).

- Can you confirm the issue when using::

  ```bash
  npm create thulite@1.7.4
  ```

- Did you clear the Hugo build directories — run for example:

  ```bash
  npx shx rm -rf public resources .hugo_build.lock hugo_stats.json
  ```

- Do you have this file present (id est are you overriding `script-footer.html`): `layouts/partials/footer/script-footer.html`?

- Did you search for `site.IsMultiLingual` in your repo — this should thus be `hugo.IsMultilingual`?

**Reply by cnichte** [reply]
Thanks for the tips. 

I work with hugo on mac that is installed ist with homebrew because I use it in other Hugo projects.

I now upgraded my "external" Version from Hugo `0.121.2` to Hugo `0.139.0`, and tadaa: its working. The error is gone - thulite apparently uses this "external" brew version.

I had always understood that Thulite is delivered with a version of hugo and only uses this. Is this still the case or is it different now? How do you control in thulite which of the Hugo versions is used - the external or the supplied one?

Best regards - Carsten

**Reply by h-enk** [reply]
Yes, you're right, previously Thulite used a locally (id est in your project) installed Hugo version (that you could control in `package.json`).

Theoretically, I still favor this setup (control all your project's dependencies in a single file,  id est in `package.json`). However, in practice, I learned that most users did not care and preferred a setup closer to Hugo's default one. So, I dropped that (making Thulite's `package.json` setup much cleaner in the process). Now Thulite is always using your globally installed Hugo version — which you can control using for example [Hugo Version Manager](https://github.com/jmooring/hvm).

I will update the Thulite docs and Doks docs to make this more clear— see also [Prerequisites](https://docs.thulite.io/install/auto/#prerequisites)
